epochStart/metachain/rewardsCreatorProxy.go

```diff
@@ -135,7 +130,7 @@ func (rcp *rewardsCreatorProxy) changeRewardCreatorIfNeeded(epoch uint32) error
 	rcp.mutRc.Lock()
 	defer rcp.mutRc.Unlock()
 
-	if epoch > rcp.epochEnableV2 {
+	if epoch > rcp.args.EnableEpochsHandler.StakingV2EnableEpoch() {
```

Comment:
not 100% sure, but you could have used `IsStakingV2FlagEnabledForActivationEpochCompleted`

Reply:
same as above, the `epoch` param comes from metablock